src/connectors/toggle-refinement/connectToggleRefinement.js

```diff
@@ -279,6 +181,124 @@ export default function connectToggleRefinement(renderFn, unmountFn = noop) {
         return state.removeDisjunctiveFacet(attribute);
       },
 
+      getRenderState(renderState, renderOptions) {
+        return {
+          ...renderState,
+          toggleRefinement: this.getWidgetRenderState(renderOptions),
+        };
+      },
+
+      getWidgetRenderState({ state, helper, results, createURL }) {
+        let value;
+
+        // results are null at init
```

Comment:
I've tried to avoid the `if:else` but I didn't find a way to make a distinction between `init` and `render` other than by using this method.